### PR TITLE
Get sysInfos based on specified platform

### PR DESCRIPTION
**Pull request by Fatme**
In case when android platform is specified, checks only android related things. In case when iOS platform is specified, checks only iOS related things. When no platform is specified, checks both Android and iOS things. This will improve the time {N} CLI needs to check if environment is correctly configured for platform specific commands. This also will fix failing cloud builds.

### CODE REVIEW

File: lib/doctor.ts [hunk 1]
```diff
@@ -32,6 +32,51 @@ export class Doctor implements NativeScriptDoctor.IDoctor {
 		let result: NativeScriptDoctor.IInfo[] = [];
 		const sysInfoData = await this.sysInfo.getSysInfo(config);
 
+		result = result.concat(
+			this.processSysInfoItem({
+				item: sysInfoData.gitVer,
+				infoMessage: "Git is installed and is configured properly.",
+				warningMessage: "Git is not installed or not configured properly.",
+				additionalInformation: "You will not be able to create and work with Screen Builder projects." + EOL
```
Comment (rosen-vladimirov) [inline lib/doctor.ts]: ughhh... what?

File: lib/doctor.ts [hunk 2]
```diff
@@ -32,6 +32,51 @@ export class Doctor implements NativeScriptDoctor.IDoctor {
...
+					+ "To be able to work with Screen Builder projects, download and install Git as described" + EOL
+					+ "in https://git-scm.com/downloads and add the git executable to your PATH.",
+				platforms: Constants.SUPPORTED_PLATFORMS
+			})
+		);
+
+		if (config && config.platform === Constants.ANDROID_PLATFORM_NAME) {
```
Comment (rosen-vladimirov) [inline lib/doctor.ts]: You can extend this:
```TypeScript
if (!config || !config.platform || config.platform === Constants.ANDROID_PLATFORM_NAME) {
```

You can do the same for iOS and this way you can remove the whole if on line 55

File: lib/doctor.ts [hunk 3]
```diff
@@ -103,7 +163,7 @@ export class Doctor implements NativeScriptDoctor.IDoctor {
 					platforms: [Constants.IOS_PLATFORM_NAME]
 				})
 			);
-
+
```
Comment (rosen-vladimirov) [inline lib/doctor.ts]: please revert this change

File: lib/doctor.ts [hunk 4]
```diff
@@ -116,7 +176,7 @@ export class Doctor implements NativeScriptDoctor.IDoctor {
 					})
 				);
 			}
-
+
```
Comment (rosen-vladimirov) [inline lib/doctor.ts]: please revert this change

File: lib/sys-info.ts [hunk 5]
```diff
@@ -9,6 +9,8 @@ import * as path from "path";
 import * as osenv from "osenv";
 import * as temp from "temp";
 import * as semver from "semver";
+import { constants } from "zlib";
```
Comment (rosen-vladimirov) [inline lib/sys-info.ts]: what's the purpose of this line?

File: test/sys-info.ts [hunk 6]
```diff
@@ -4,6 +4,7 @@ import { EOL } from "os";
 import { SysInfo } from "../lib/sys-info";
 import { Helpers } from "../lib/helpers";
 import { ChildProcess } from "../lib/wrappers/child-process";
+import { constants } from "zlib";
```
Comment (rosen-vladimirov) [inline test/sys-info.ts]: what's the purpose of this line?

File: lib/doctor.ts (hunk 1 above)
Comment (Fatme) [reply lib/doctor.ts]: Legacy check :smile: will remove it

File: lib/sys-info.ts (hunk 5 above)
Comment (Fatme) [reply lib/sys-info.ts]: Wrong copy paste

File: test/sys-info.ts (hunk 6 above)
Comment (Fatme) [reply test/sys-info.ts]: Wrong copy paste :smile:
